src/Security/BasicAuth.php

```diff
@@ -71,33 +77,40 @@ public static function requireLogin($realm, $permissionCode = null, $tryUsingSes
 		 * The follow rewrite rule must be in the sites .htaccess file to enable this workaround
 		 * RewriteRule .* - [E=HTTP_AUTHORIZATION:%{HTTP:Authorization}]
 		 */
-        $authHeader = (isset($_SERVER['HTTP_AUTHORIZATION']) ? $_SERVER['HTTP_AUTHORIZATION'] :
-                  (isset($_SERVER['REDIRECT_HTTP_AUTHORIZATION']) ? $_SERVER['REDIRECT_HTTP_AUTHORIZATION'] : null));
+        $authHeader = $request->getHeader('Authorization');
         $matches = array();
         if ($authHeader && preg_match('/Basic\s+(.*)$/i', $authHeader, $matches)) {
             list($name, $password) = explode(':', base64_decode($matches[1]));
-            $_SERVER['PHP_AUTH_USER'] = strip_tags($name);
-            $_SERVER['PHP_AUTH_PW'] = strip_tags($password);
+            $request->addHeader('PHP_AUTH_USER', strip_tags($name));
+            $request->addHeader('PHP_AUTH_PW', strip_tags($password));
         }
 
         $member = null;
-        if (isset($_SERVER['PHP_AUTH_USER']) && isset($_SERVER['PHP_AUTH_PW'])) {
-            $member = MemberAuthenticator::authenticate(array(
-                'Email' => $_SERVER['PHP_AUTH_USER'],
-                'Password' => $_SERVER['PHP_AUTH_PW'],
-            ), null);
+
+        if ($request->getHeader('PHP_AUTH_USER') && $request->getHeader('PHP_AUTH_PW')) {
+            /** @var Authenticator $authenticator */
+            $authenticator = Injector::inst()->get(Authenticator::class);
+
+            $member = $authenticator->authenticate([
+                'Email' => $request->getHeader('PHP_AUTH_USER'),
+                'Password' => $request->getHeader('PHP_AUTH_PW'),
+            ]);
```

Comment:
Seconded; Please make this method nullable (you can have a by-ref nullable argument).